hpat/tests/tests_perf/test_perf_utils.py

```diff
@@ -0,0 +1,140 @@
+#!/usr/bin/env python
+# -*- coding: utf-8 -*-
+#
+# *****************************************************************************
+# Copyright (c) 2019, Intel Corporation All rights reserved.
+#
+# Redistribution and use in source and binary forms, with or without
+# modification, are permitted provided that the following conditions are met:
+#
+#     Redistributions of source code must retain the above copyright notice,
+#     this list of conditions and the following disclaimer.
+#
+#     Redistributions in binary form must reproduce the above copyright notice,
+#     this list of conditions and the following disclaimer in the documentation
+#     and/or other materials provided with the distribution.
+#
+# THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS"
+# AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO,
+# THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR
+# PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDER OR
+# CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL,
+# EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO,
+# PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS;
+# OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY,
+# WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR
+# OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE,
+# EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+# *****************************************************************************
+
+import sys
+import pandas
+
+"""
+Utility functions collection to support performance testing of
+functions implemented in the project
+
+Data generators:
+    perf_data_gen() generates list of items with fixed length
+
+Data handling:
+    add_results() add an experiment timing results to globla storage
+    print_results() print all timing results from global storage
+
+"""
+
+
+def perf_data_gen(tmpl, max_item_len, max_bytes_size):
+    """
+    Data generator produces 2D like data.
+                  tmpl: list of input template string
+          max_item_len: length (in elements) of resulted string in an element of the result array
+        max_bytes_size: maximum size in bytes of the return data
+
+                return: list of strings
+    """
+
+    result = []
+    obj_size = sys.getsizeof(tmpl, -1)
+
+    while (obj_size < max_bytes_size) and (obj_size >= 0):
+        for item in tmpl:
+            local_item = item
+            local_item_len = len(local_item)
+
+            while (local_item_len < max_item_len) and (local_item_len >= 0):
+                local_item += item
+                local_item_len = len(local_item)
+
+            result.append(local_item)
+
+        obj_size = sys.getsizeof(result, -1)
```

Comment:
@shssf here we get size of list in bytes. I think this way is incorrect, because here we get size of list of pointers in bytes, but not list of pointers + strings inside of the list. E.g.:
>>> sys.getsizeof(['hello'])
72
>>> sys.getsizeof(['hello', 'first_world'])
80
>>> sys.getsizeof(['hello', 'first_world', 'and', 'second_world'])
96

Here pointer to string takes 8 bytes, but we don't consider size of each string in the list.
So I propose to use the following function to get real size of list:
```
def get_size(obj):
    """Sum size of object and its members."""
    size = 0
    processed_ids = set()
    objects = [obj]
    while objects:
        need_refer = []
        for obj in objects:
            if id(obj) in processed_ids:
                continue
            processed_ids.add(id(obj))
            need_refer.append(obj)
            size += sys.getsizeof(obj)
        objects = gc.get_referents(*need_refer)
    return size
```
>>> get_size(['hello'])
126
>>> get_size(['hello', 'first_world'])
194
>>> get_size(['hello', 'first_world', 'and', 'second_world'])
323

Please correct me if I'm wrong. If you agree with me, I will make the change.